File: Basic Programming/Hackerrank 30 days of code c++ solution/Day_15.cpp
/*                    Name : Arif Khan
                      Judge: HACKERRANK
                      University: Primeasia University
                      problem: Day 15: Linked List
                      Difficulty: Easy
                      Problem Link: https://www.hackerrank.com/challenges/30-linked-list/problem
*/

#include <iostream>
#include <cstddef>
using namespace std;

class Node
{
    public:
        int data;
        Node *next;
        Node(int d){
            data=d;
            next=NULL;
        }
};
class Solution{
    public:

      Node* insert(Node *head,int data)
      {
          Node* newNode = new Node(data);
          Node * tail = head;

          if(!head)
             return newNode;
          for(;tail->next; tail = tail->next);
          tail->next = newNode;

          retur head;
      }

      void display(Node *head)
      {
          Node *start=head;
          while(start)
          {
              cout<<start->data<<" ";
              start=start->next;
          }
      }
};

int main()
{
	Node* head=NULL;
  	Solution mylist;
    int T,data;
    cin>>T;
    while(T-->0){
        cin>>data;
        head=mylist.insert(head,data);
    }	
	mylist.display(head);
		
}

File: Basic Programming/Hackerrank 30 days of code python3 solution/Day_15.py
#                      Name : Arif Khan
#                      Judge: HACKERRANK
#                      University: Primeasia University
#                      problem: Day 15: Linked List
#                      Difficulty: Easy
#                      Problem Link: https://www.hackerrank.com/challenges/30-linked-list/problem
#

class Node:
    def __init__(self,data):
        self.data = data
        self.next = None 
class Solution: 
    def display(self,head):
        current = head
        while current:
            print(current.data,end=' ')
            current = current.next

    def insert(self,head,data): 
        n = Node(data)
        if head:
            current = head
            while current.next:
                current = current.next
            current.next = n
            return head
        else:
            return n
    #Complete this method

mylist= Solution()
T=int(input())
head=None
for i in range(T):
    data=int(input())
    head=mylist.insert(head,data)    
mylist.display(head); 	  